app/screens/global_threads/threads_list/thread/thread.tsx

```diff
@@ -0,0 +1,239 @@
+            overflow: 'hidden',
+        },
+        threadDeleted: {
+            color: changeOpacity(theme.centerChannelColor, 0.72),
+            fontStyle: 'italic',
+        },
+        threadStarter: {
+            color: theme.centerChannelColor,
+            fontSize: 15,
+            fontWeight: '600',
+            lineHeight: 22,
+            paddingRight: 8,
+        },
+        channelNameContainer: {
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.08),
+            borderRadius: 4,
+            maxWidth: '50%',
+        },
+        channelName: {
+            color: theme.centerChannelColor,
+            ...typography('Body', 25, 'SemiBold'),
+            letterSpacing: 0.1,
+            textTransform: 'uppercase',
+            marginHorizontal: 12,
+            marginVertical: 2,
+        },
+        date: {
+            color: changeOpacity(theme.centerChannelColor, 0.64),
+            ...typography('Body', 25, 'Light'),
+        },
+        message: {
+            color: theme.centerChannelColor,
+            fontSize: 15,
+            lineHeight: 20,
+        },
+        unreadDot: {
+            width: 8,
+            height: 8,
+            borderRadius: 4,
+            backgroundColor: theme.sidebarTextActiveBorder,
+            alignSelf: 'center',
+            marginTop: 5,
+        },
+        mentionBadge: {
+            width: 18,
+            height: 18,
+            borderRadius: 9,
+            backgroundColor: theme.buttonBg,
+            alignSelf: 'center',
+        },
+        mentionBadgeText: {
+            ...typography('Body', 25, 'SemiBold'),
+            alignSelf: 'center',
+            color: theme.buttonColor,
+        },
+    };
+});
+
+const Thread = ({author, channel, post, teammateNameDisplay, testID, theme, thread}: Props) => {
+    const intl = useIntl();
+    const isTablet = useIsTablet();
+    const styles = getStyleSheet(theme);
+    const serverUrl = useServerUrl();
+
+    const showThread = useCallback(preventDoubleTap(() => {
+        fetchAndSwitchToThread(serverUrl, thread.id);
+    }), [serverUrl, thread.id]);
+
+    const showThreadOptions = useCallback(() => {
+        const passProps = {thread};
+        const title = isTablet ? intl.formatMessage({id: 'thread.options.title', defaultMessage: 'THREAD ACTIONS'}) : '';
+
+        if (isTablet) {
+            showModal(Screens.THREAD_OPTIONS, title, passProps, bottomSheetModalOptions(theme, 'close-thread-options'));
+        } else {
+            showModalOverCurrentContext(Screens.THREAD_OPTIONS, passProps);
+        }
+    }, [isTablet, theme, thread]);
+
+    const threadStarterName = displayUsername(author, intl.locale, teammateNameDisplay);
+    const testIDPrefix = `${testID}.${thread.id}`;
+
+    const needBadge = thread.unreadMentions || thread.unreadReplies;
+    let badgeComponent;
+    if (needBadge) {
+        if (thread.unreadMentions && thread.unreadMentions > 0) {
+            badgeComponent = (
```

Comment:
Just a wild thought.  `needBadge` checks for `unreadMentions` or `unreadReplies`.  Then the `badgeComponent` would either  display `unreadMentions` or `unreadReplies`.  But can both co-exist?

Reply:
No, they don't co-exist :) The design has space for only one and the `mentions` get the priority.